tests/test_warc_metadata_sidecar.py

```diff
@@ -136,13 +144,15 @@ def test_metadata_sidecar_dns_record(self, mock_warcwriter, caplog, tmpdir):
 
     @patch('warc_metadata_sidecar.find_character_set')
     @patch('warc_metadata_sidecar.find_language')
-    def test_metadata_sidecar_image_record(self, mock_language, mock_character, tmpdir):
+    @patch('warc_metadata_sidecar.determine_soft404')
+    def test_metadata_sidecar_image_record(self, mock_404, mock_language, mock_character, tmpdir):
         img_payload = '{0} {1}\n{2} {3}'.format(
                             sidecar.MIME_TITLE, {'fido': 'image/gif', 'python-magic': 'image/gif'},
                             sidecar.PUID_TITLE, 'fmt/4').encode('utf-8')
         sidecar.metadata_sidecar(str(tmpdir), IMAGE_TEST_FILE)
         mock_language.assert_not_called()
         mock_character.assert_not_called()
+        mock_404.assert_not_called()
```

Comment:
I like that you added this check here. We could similarly check that `determine_soft404` is called a couple tests up for `test_metadata_sidecar`.